### PR TITLE
Update "focusable element has no keyboard trap" examples to not use single key shortcuts (80af7b)

### LINKED ISSUE #1012: Rule 8 "Focusable element has no keyboard trap" (80af7b) uses single character key to ESC
The rule has trapped examples where pressing a key escapes the trap (Press the M-key to Exit). You may want to update that to something like "Press STR-M to Exit" (or even use ESC key?)  for it to be in clearly line with SC 21.4 (although use ot as single character key may fall under the exception "Active only on focus" - haven't rendered the examples to work that out)

### PR DESCRIPTION
**Pull request by carlosapaduarte**
Updates the examples of the composite rule "Focusable element has no keyboard trap" ([80af7b](https://act-rules.github.io/rules/80af7b)) (together with the atomic rules "Focusable element has no keyboard trap via standard navigation" ([a1b64e](https://act-rules.github.io/rules/a1b64e)) and "Focusable element has no keyboard trap via non-standard navigation" ([ebe86a](https://act-rules.github.io/rules/ebe86a))) so that they do not use single key shortcuts to make them comply with [SC 2.1.4](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html).

Closes issue(s):

- closes #1012 

Need for Call for Review:
This will require a 2 weeks Call for Review

---

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Call for Review period. In case of disagreement, the longer period wins.

### CODE REVIEW

**Review by Jym77 — DISMISSED** [review]

**Review by WilcoFiers — DISMISSED** [review]

File: _rules/focusable-no-keyboard-trap-80af7b.md [hunk 1]
```diff
@@ -68,45 +68,45 @@ _There are no major accessibility support issues known for this rule._
...
-Keyboard trap with help information in a paragraph before, and where the method advised works.
+These focusable `button` elements create a keyboard trap with help information in a paragraph before, and where the method advised works.
 
 ```html
 <script>
 	var trapOn = false
 </script>
 
-<p>Press the M-key to Exit</p>
+<p>Press Ctrl+M to Exit</p>
```
[the remark this answers is not in the document]
Comment (carlosapaduarte) [reply _rules/focusable-no-keyboard-trap-80af7b.md:102]: The Ctrl+M shortcut also works on Mac

File: _rules/focusable-no-keyboard-trap-80af7b.md [hunk 2]
```diff
@@ -140,18 +140,20 @@ Keyboard trap with help information within the trap, and where the method advise
...
 
 	function showHelpText() {
-		document.getElementById('helptext').innerHTML = '<p>Press the M-key to Exit</p>'
+		document.getElementById('helptext').innerHTML = '<p>Press Ctrl+M to Exit</p>'
 	}
 </script>
 
-<div onkeydown="(function(e){ if (e.keyCode === 77){trapOn=false;document.getElementById('link2').focus();}})(event)">
+<div
+	onkeydown="(function(e){ if (e.keyCode === 77 && e.ctrlKey){trapOn=false;document.getElementById('link2').focus();}})(event)"
```
[the remark this answers is not in the document]
Comment (carlosapaduarte) [reply _rules/focusable-no-keyboard-trap-80af7b.md:155]: Not really sure... In this instance I believe having the if conditions visible next to the event could be important to understand when the trap is created or skipped. If we "hide" that in a script file (or move it inside a script tag) we might be decreasing readability. What are others' thoughts on this?

File: _rules/focusable-no-keyboard-trap-80af7b.md [hunk 3]
```diff
@@ -68,45 +68,45 @@ _There are no major accessibility support issues known for this rule._
...
+These [focusable][] elements do not create a trap for keyboard navigation.
 
 ```html
 <a href="#">Link 1</a> <button>Button1</button>
 ```
 
 #### Passed Example 2
 
-Using `tabindex="1"`.
+This [focusable][] element does not create a trap for keyboard navigation.
```
Comment (WilcoFiers) [inline _rules/focusable-no-keyboard-trap-80af7b.md:79]: I don't think example descriptions should repeat. That suggests to me they're not specific enough.

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: _rules/focusable-no-keyboard-trap-80af7b.md (hunk 3 above)
Comment (carlosapaduarte) [reply _rules/focusable-no-keyboard-trap-80af7b.md]: Revised the descriptions

File: _rules/focusable-no-keyboard-trap-80af7b.md [hunk 4]
```diff
@@ -56,6 +54,8 @@ _There are no major accessibility support issues known for this rule._
 
 ## Background
 
+This rule only applies to HTML and SVG. Thus, it is a partial check for WCAG 2.0 success criterion 2.1.2, which applies to all content.
```
Comment (Jym77) [inline _rules/focusable-no-keyboard-trap-80af7b.md:57]: I'm wondering whether we should completely remove this note instead.
It is essentially true for every single rule we have that it applies only to HTML and sometimes SVG and therefore is a partial check for WCAG… But we're not putting the note in every rule. Why would this rule specifically need it?

(also if kept we should probably refer to WCAG 2.1 and link to the SC)

File: _rules/focusable-no-keyboard-trap-non-standard-nav-ebe86a.md [hunk 5]
```diff
@@ -54,6 +52,8 @@ There are no major accessibility support issues known for this rule.
 
 ## Background
 
+This rule only applies to HTML and SVG. Thus, it is a partial check for WCAG 2.0 success criterion 2.1.2, which applies to all content.
```
Comment (Jym77) [inline _rules/focusable-no-keyboard-trap-non-standard-nav-ebe86a.md:55]: Same remark.

File: _rules/focusable-no-keyboard-trap-80af7b.md [hunk 6]
```diff
@@ -76,91 +76,120 @@ These [focusable][] elements do not create a trap for keyboard navigation.
...
 #### Passed Example 4
 
-These focusable `button` elements create a keyboard trap with help information in a paragraph before, and where the method advised works.
+These focusable `button` elements have scripts that create a keyboard trap. The document includes help information in a paragraph before the `button` elements and the method advised works to escape the keyboard trap.
 
 ```html
 <script>
 	var trapOn = false
+
+	const moveFocusToButton = btn => {
```
Comment (Jym77) [inline _rules/focusable-no-keyboard-trap-80af7b.md:101]:
```suggestion
	function moveFocusToButton(btn) {
```
might be a bit more readable.

File: _rules/focusable-no-keyboard-trap-80af7b.md [hunk 7]
```diff
@@ -76,91 +76,120 @@ These [focusable][] elements do not create a trap for keyboard navigation.
...
+		if (trapOn) {
+			document.getElementById(btn).focus()
+		}
+	}
+
+	const escapeTrapOnCtrlM = e => {
```
Comment (Jym77) [inline _rules/focusable-no-keyboard-trap-80af7b.md:107]:
```suggestion
	function escapeTrapOnCtrlM(e) {
```

Comment (Jym77) [inline _rules/focusable-no-keyboard-trap-80af7b.md:107]: These two functions are used in many examples in two rules and might benefit from a shared asset. Obviously with the drawback of less locality… (that might be solved by #1805 ).

File: _rules/focusable-no-keyboard-trap-standard-nav-a1b64e.md [hunk 8]
```diff
@@ -44,6 +42,8 @@ There are no major accessibility support issues known for this rule.
 
 ## Background
 
+This rule only applies to HTML and SVG. Thus, it is a partial check for WCAG 2.0 success criterion 2.1.2, which applies to all content.
```
Comment (Jym77) [inline _rules/focusable-no-keyboard-trap-standard-nav-a1b64e.md:45]: Same remark.

File: _rules/focusable-no-keyboard-trap-standard-nav-a1b64e.md [hunk 9]
```diff
@@ -82,7 +82,7 @@ This [focusable][] element does not create a trap for keyboard navigation.
 
 #### Failed Example 1
 
-This [focusable][] element creates a keyboard trap.
+This [focusable][] element creates a keyboard trap brining focus to the `button`.
```
Comment (Jym77) [inline _rules/focusable-no-keyboard-trap-standard-nav-a1b64e.md:85]:
```suggestion
This [focusable][] element creates a keyboard trap bringing focus to the `button`.
```

**Review by Jym77 — DISMISSED** [review]

File: _rules/focusable-no-keyboard-trap-80af7b.md (hunk 4 above)
Comment (carlosapaduarte) [reply _rules/focusable-no-keyboard-trap-80af7b.md:57]: Agree that there is nothing to make this rule special. I'm removing it.

File: _rules/focusable-no-keyboard-trap-80af7b.md (hunk 7 above)
Comment (carlosapaduarte) [reply _rules/focusable-no-keyboard-trap-80af7b.md:107]: Hoping that #1805 will become a reality soon, I'm moving all scripts to test-assets

File: _rules/focusable-no-keyboard-trap-80af7b.md [hunk 10]
```diff
@@ -68,97 +66,79 @@ _There are no major accessibility support issues known for this rule._
...
+<button id="btn2" onfocus="trapOn = true" onblur="moveFocusToButton('btn1')" onkeydown="escapeTrapOnCtrlM(event)">
 	Button 2
 </button>
 <a id="link2" href="#">Link 2</a>
 ```
 
 #### Passed Example 6
 
-Keyboard trap with "help" link that once clicked exposes the instructions.
+These focusable `button` elements have scripts that create a keyboard trap. The document includes help information in an "help" link that once clicked exposes the instructions to escape the keyboard trap.
```
Comment (kengdoj) [inline _rules/focusable-no-keyboard-trap-80af7b.md:129]:
```suggestion
These focusable `button` elements have scripts that create a keyboard trap. The document includes help information in a "help" link that once clicked exposes the instructions to escape the keyboard trap.
```

File: _rules/focusable-no-keyboard-trap-non-standard-nav-ebe86a.md [hunk 11]
```diff
@@ -66,73 +64,55 @@ There are no major accessibility support issues known for this rule.
...
+<button id="btn2" onfocus="trapOn = true" onblur="moveFocusToButton('btn1')" onkeydown="escapeTrapOnCtrlM(event)">
 	Button 2
 </button>
 <a id="link2" href="#">Link 2</a>
 ```
 
 #### Passed Example 3
 
-Keyboard trap with "help" link that once clicked exposes the instructions.
+These focusable `button` elements have scripts that create a keyboard trap. The document includes help information in an "help" link that once clicked exposes the instructions to escape the keyboard trap.
```
Comment (kengdoj) [inline _rules/focusable-no-keyboard-trap-non-standard-nav-ebe86a.md:103]:
```suggestion
These focusable `button` elements have scripts that create a keyboard trap. The document includes help information in a "help" link that once clicked exposes the instructions to escape the keyboard trap.
```

**Review by kengdoj — APPROVED** [review]
Just editorial changes of "an" to "a".

**Review by wcreedle — APPROVED** [review]

**Comment by carlosapaduarte** [comment]
Call for review ends April 14

**Comment by carlosapaduarte** [comment]
Call for review ended. Merging.

**Pull request merged**